lib/client.js

```diff
@@ -119,8 +137,11 @@ function wrappedPostMessage (name, params) {
     rawPostMessage(this, msg)
   })
 
-  // ensure promise is cleaned up when resolved
-  return promise.then(removePromise.bind(null, id, timeoutId), removePromise.bind(null, id, timeoutId))
+  // ensure promise is cleaned up when resolved/rejected, pass the start time for resolved promises
+  return promise.then(
+    removePromise.bind(null, id, timeoutId, this, actions, promiseStartTime),
```

Comment:
Another silly question: how does bind work in this instance?  I see that from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind that `null` would take the roll of `args` in the function itself - is that correct?

Reply:
It may be easier to explain in person, but in this case we're:
1) keeping the value of `this` when calling this function by setting the first argument to `null`,
2) "pre-setting" the value of the first 5 params in the `removePromise` call with arguments 2-6 in these bound functions,
3) using the returned value from the wrapping `promise.then` call as the remaining `args` when calling `removePromise`.

So when `removePromise` is called it should always have an `args` value depending on whether the wrapping promise was resolved/rejected

Reply:
Thanks for the detailed and helpful explanation!  I'm still a little bit confused on the first point, maybe we could discuss that later?